src/main/java/com/github/mproberts/rxtools/map/SubjectMap.java

```diff
@@ -331,7 +331,64 @@ public void runIfBound(K key, final Runnable runnable)
         }
 
         if (isBound) {
-            runnable.run();
+            processNewFaultForKey(key);
+        }
+    }
+
+    private void processNewFaultForKey(final K key)
+    {
+        Completable.defer(new Callable<CompletableSource>() {
+            @Override
+            public CompletableSource call() throws Exception {
+                return new CompletableSource() {
+                    @Override
+                    public void subscribe(CompletableObserver completableObserver) {
+                        Function<K, Single<V>> faultHandler = _faultHandler;
+
+                        emitFault(key);
+
+                        if (faultHandler != null) {
+                            try {
+                                Single<V> fault = faultHandler.apply(key);
+
+                                fault.doOnSuccess(new Consumer<V>() {
+                                    @Override
+                                    public void accept(V v) throws Exception {
+                                        WeakReference<Processor<V, V>> weakSource = _weakSources.get(key);
+                                        Processor<V, V> processor;
+                                        if (weakSource != null && (processor = weakSource.get()) != null ) {
+                                            processor.onNext(v);
+                                        }
+                                    }
+                                }).toCompletable().subscribe(completableObserver);
+                            } catch (Exception e) {
+                                completableObserver.onError(e);
+                            }
+                        } else {
+                            completableObserver.onComplete();
+                        }
+                    }
+                };
+            }
+        }).subscribe();
+    }
+
+    /**
+     * Re-emits a fault for all bound keys 
+     */
+    public void faultAllBound()
+    {
+        List<K> keys = new ArrayList<>();
```

Comment:
do you have to create a new array list? Does `keySet` not return a copy of the key list?

And if you really have to, maybe we could initialize the list with the right size